doc/python/sphinx/glossary.rst

```diff
@@ -39,3 +39,11 @@ Glossary
     until the channel is READY. Any submitted RPCs may still fail before the
     READY state is reached for other reasons, e.g., the client channel has been
     shut down or the RPC's deadline has been reached.
+
+  channel_arguments
```

Comment:
So this is tricky. Channel arguments are (in some cases) intentionally undocumented because they are not stable APIs. We need to make clear here that channel arguments may constitute experimental APIs.

Reply:
Add "contain experimental API" to the text. To me, if we say it is both unstable and advanced, it sounds like in a really bad shape. WDYT?

Reply:
Agree. But better safe than sorry. We can get an opinion from the Core team, since channel args are really a passthrough to a set of their APIs.

@markdroth What is the Core team's stance on the API stability of channel arguments?

To soften the statement, I'd suggest "s/and contain experimental API/and individual keys may constitute experimental APIs/".

Reply:
Unfortunately, this is a bit of a mess today.

Historically, we've not done a great job of setting expectations here.  We've added a lot of channel args that were intended to be experimental but not actually marked as such.  So some of them are fairly stable and reliable, and others are less so, and there's no way for users to tell the difference, which sucks.

To make matters worse, even for the stable options, I don't think we ever made a conscious design decision about how to expose these options in wrapped languages.  I don't think we ever intended the literal option strings to become part of the wrapped language APIs, but that's where we've wound up, because we never established any other design principle.

Ideally, I would like to see us move to a place where the wrapped language APIs provide a different (cleaner) API for specifying the options that we do consider stable and reliable.  Direct access to the channel args API should be used only as an escape hatch for the experimental args, and those should never be considered stable.

I suspect that in order to get from here to there, we will need a fair amount of work.  We will need to go through the current set of channel args, decide which ones are stable and reliable, agree on what semantics we want to use to expose those options in wrapped languages, and then implement those semantics in each language.  Probably not going to happen overnight, but if you feel like driving this forward, it would be a big win.

Sorry I don't have a better response for you.

Reply:
Updated our promise to:
```
Channel arguments are meant for advanced usages and contain experimental API (some may not labeled as experimental).
```

@markdroth Thanks for sharing the context around the channel arguments. Given the unstable nature of channel arguments, we do want to warn our users. (I checked the history of the `grpc_types.h`, and we do update the key name from time to time.)

Standardize the channel arguments is possible. For Java & Golang, they provide some channel arguments via surface API. We should be able to unify a stable set of channel arguments that recommend all wrapper languages to expose as public API. I don't mind add this to my todo list.

This PR still provide value to Python users, if they do want advanced features. Do you think this version is better? @gnossen

Reply:
Works for me. Long term, we should try to implement a Python veneer for the most important channel args as Mark suggested. We've already done this for some things, like compression. It would make sense to do the same for others in the future, like "max_receive_message_length". For now, let's just keep it in mind for the next time planning comes around.